genie-proto/src/main/proto/genie.proto

```diff
@@ -264,3 +264,40 @@ message JobKillRegistrationRequest {
 
 message JobKillRegistrationResponse {
 }
+
+//--------------------------------------------------------------------
+// FileStream Service
+//
+// This service streams file local to an agent back to the server.
+//--------------------------------------------------------------------
+
+service FileStreamService {
+    rpc sync (stream AgentManifestMessage) returns (stream ServerControlMessage);
+    rpc transmit (stream AgentFileMessage) returns (stream ServerAckMessage);
+}
+
+message AgentManifestMessage {
+    string job_id = 1;
+    string manifest_json = 2;
+}
+
+message ServerControlMessage {
+    oneof message {
```

Comment:
You have a one of here but no other message. Seems an unnecessary wrapper?

Reply:
Thought i would "future-proof" this particular stream (from server to agent) protocol, in case later we came up with another kind of 'control' message.

If i don't make this a `oneof` right now, and later we want to add a message, we need to throw away this whole RPC service definition and start anew.